monitors.go

```diff
@@ -35,54 +37,79 @@ type Monitor struct {
 
 // Polls the scheduler every certain amount of time to see if the update has succeeded
 func (m *Monitor) JobUpdate(updateKey aurora.JobUpdateKey, interval int, timeout int) (bool, error) {
+	status, err := m.JobUpdateStatus(updateKey,
+		map[aurora.JobUpdateStatus]bool{
+			aurora.JobUpdateStatus_ROLLED_FORWARD: true,
+			aurora.JobUpdateStatus_ROLLED_BACK:    true,
+			aurora.JobUpdateStatus_ABORTED:        true,
+			aurora.JobUpdateStatus_ERROR:          true,
+			aurora.JobUpdateStatus_FAILED:         true,
+		},
+		time.Duration(interval)*time.Second,
+		time.Duration(timeout)*time.Second)
+
+	if err != nil {
+		return false, err
+	}
+
+	// Rolled forward is the only state in which an update has been successfully updated
+	// if we encounter an inactive state and it is not at rolled forward, update failed
+	switch status {
+	case aurora.JobUpdateStatus_ROLLED_FORWARD:
+		m.Client.RealisConfig().logger.Println("update succeeded")
+		return true, nil
+	case aurora.JobUpdateStatus_ROLLED_BACK:
+		m.Client.RealisConfig().logger.Println(RolledBack)
+		return false, errors.New(RolledBack)
+	case aurora.JobUpdateStatus_ABORTED:
+		m.Client.RealisConfig().logger.Println(UpdateAborted)
+		return false, errors.New(UpdateAborted)
+	case aurora.JobUpdateStatus_ERROR:
+		m.Client.RealisConfig().logger.Println(UpdateError)
+		return false, errors.New(UpdateError)
+	case aurora.JobUpdateStatus_FAILED:
+		m.Client.RealisConfig().logger.Println(UpdateFailed)
+		return false, errors.New(UpdateFailed)
+	default:
+		return false, nil
+	}
+}
+
+func (m *Monitor) JobUpdateStatus(updateKey aurora.JobUpdateKey, desiredStatuses map[aurora.JobUpdateStatus]bool, interval, timeout time.Duration) (aurora.JobUpdateStatus, error) {
```

Comment:
Considering breaking this long line into several shorter ones to make the code a bit easier to read

Reply:
Fixed!